src/code/audio/code_8019AF00.c

```diff
@@ -4463,35 +4744,238 @@ void Audio_ClearSariaBgm2(void) {
+                sPrevMainBgmSeqId = sPrevAmbienceSeqId;
+            }
+            Audio_QueueSeqCmd(0x00000000 | ((u32)(SEQ_PLAYER_BGM_MAIN) << 24) | ((u32)(0) << 0x10) |
+                              (u32)(sPrevMainBgmSeqId + 0x8000));
+        }
+        sPrevMainBgmSeqId = NA_BGM_DISABLED;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/Audio_IsSequencePlaying.s")
+// Unused
+void Audio_PlayAmbience_StorePrevBgm(u8 ambienceId) {
+    u16 seqId = Audio_GetActiveSequence(SEQ_PLAYER_BGM_MAIN);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A2E54.s")
+    if (seqId != NA_BGM_AMBIENCE) {
+        sPrevMainBgmSeqId = seqId;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A2ED8.s")
+    Audio_PlayAmbience(ambienceId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A2F88.s")
+// Unused
+void Audio_ForceRestorePreviousBgm(void) {
+    if (sPrevMainBgmSeqId != NA_BGM_DISABLED) {
+        Audio_QueueSeqCmd(0x00000000 | ((u32)(SEQ_PLAYER_BGM_MAIN) << 24) | ((u32)(0) << 0x10) |
+                          (u32)(sPrevMainBgmSeqId + 0x8000));
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A2FC4.s")
+    sPrevMainBgmSeqId = NA_BGM_DISABLED;
+}
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A3000.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A3038.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A3098.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/Audio_PlayFanfare.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A312C.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A31EC.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A3238.s")
+void Audio_PlaySequenceWithSeqPlayerIO(s8 playerIndex, u16 seqId, u8 fadeInDuration, s8 ioPort, u8 ioData) {
+    u16 flaggedSeqId;
+
+    Audio_QueueSeqCmd(0x70000000 | ((u32)(playerIndex) << 24) | ((u32)(ioPort) << 0x10) | (u32)(ioData));
+    if ((seqId & 0xFF) < 2) {
+        flaggedSeqId = seqId;
+    } else {
+        flaggedSeqId = seqId | 0x8000;
+    }
+    Audio_QueueSeqCmd(0x00000000 | ((u32)(playerIndex) << 24) | ((u32)(fadeInDuration) << 0x10) | (u32)(flaggedSeqId));
+}
+
+void Audio_SetSequenceMode(u8 seqMode) {
+    s32 volumeFadeInTimer;
+    u16 seqId;
+    u8 volumeFadeOutTimer;
+
+    if ((sPrevMainBgmSeqId == NA_BGM_DISABLED) && (sPrevMainBgmSeqId == NA_BGM_DISABLED)) {
+        // clang-format off
+        if (sAudioCutsceneFlag || sSpatialSeqIsActive[SEQ_PLAYER_BGM_SUB]) { \
+            seqMode = SEQ_MODE_IGNORE;
+        }
+        // clang-format on
+
+        seqId = gActiveSeqs[SEQ_PLAYER_BGM_MAIN].seqId;
+
+        if ((seqId == NA_BGM_DISABLED) || (sSeqFlags[(u8)(seqId & 0xFF)] & SEQ_FLAG_ENEMY) ||
+            ((sPrevSeqMode & 0x7F) == SEQ_MODE_ENEMY)) {
+            if (seqMode != (sPrevSeqMode & 0x7F)) {
```

Comment:
Guessing these will get macros too.

Reply:
Yeah, that can come with the audioSeqCmd PR, where the `0x80`/`0x7F` is properly documented